federationtypes.go

```diff
@@ -429,27 +369,34 @@ func (r RespSendJoin) Check(ctx context.Context, keyRing JSONVerifier, joinEvent
 		return err
 	}
 
-	stateEventsByID := map[string]*Event{}
-	authEvents := NewAuthEvents(nil)
+	authEventsByID := map[string]*Event{}
+	stateEvents := NewAuthEvents(nil)
```

Comment:
This is a confusing name surely.

Reply:
Why rename this? `authEventProvider` would be clearer given comments on `:448`.

Reply:
I've named it `authEventProvider`.